### ISSUE TITLE
Limit the amount of posts that may be fetched with GET /api/v3/posts request

### ISSUE THREAD

**Issue by tuxpiper** [ushahidi/platform#3939]
**Which user group(s) are the primary audience for this feature/enhancement.**
This was initially reported by a deployer with significant engagement. They handle personal data that is made available publicly, and it's not an option to hide it completely as sharing data is how that deployment works.

**Is your feature request related to a problem? Please describe.**
They were seeing how others where API-scraping their published posts for unknown purposes.

While they could rate limit the requests on an IP-address basis, these efforts were hampered by the fact that the requesting party could just ask for more posts per request.

**Describe the solution you'd like**

* Unprivileged requests should have a default maximum (i.e. 20) in how many posts they can fetch in one go
* Privileged requests should either be unlimited or have a different higher limit

What constitutes a privileged request? I think it would have to be either associated to a user permission ("Manage Posts"?) or to an API key permission/scope

This limit should **not** affect other endpoints that deal with post data, such as /api/v3/geojson . 

By nature those endpoints discard most of the post data and may even do further processing of the data with the effect of aggregating it and further reducing the amount of individual post detail exposed.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

Aha! Link: https://ushahiditeam.aha.io/features/PROD-123

**Comment by rowasc** [comment]
regarding privileged: I think the way we do it for exports is through permissions     with DATA_IMPORT_EXPORT so that would fit the same pattern... altough it may be possible that a scope would have been the correct decision to be honest 🤔  
From slack chat
>I think last week my conclusion is that scopes generally would be a better fit when we are doing resource actions matched to scopes rather than this in particular (unless we consider it a separate action) because we would need to do some weird stuff with ?querystrings in our scope checking middleware to even handle it if we follow the ?limit= offset thing.. since we apply the middlewares in the routes (and you can't assign a middleware through checking querystrings in routes.php/whatever without some hacky hacks)

**Comment by wino** [comment]
@tuxpiper @rowasc I took a crack at this one and have a forked PR here: https://github.com/ushahidi/platform/pull/4142
I went with the "Manage Posts" permission mentioned above for now. I think it could be switched to use scopes if needed, but I didn't have time today to figure it out in my first crack at this project.
I also tried to look into writing a test but didn't quite figure out how to handle the auth. I could give that another shot next week if needed.
